iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/relational/tvf/CumulateTableFunction.java

```diff
@@ -0,0 +1,133 @@
+package org.apache.iotdb.commons.udf.builtin.relational.tvf;
+
+import org.apache.iotdb.udf.api.exception.UDFException;
+import org.apache.iotdb.udf.api.relational.TableFunction;
+import org.apache.iotdb.udf.api.relational.access.Record;
+import org.apache.iotdb.udf.api.relational.table.TableFunctionAnalysis;
+import org.apache.iotdb.udf.api.relational.table.TableFunctionProcessorProvider;
+import org.apache.iotdb.udf.api.relational.table.argument.Argument;
+import org.apache.iotdb.udf.api.relational.table.argument.DescribedSchema;
+import org.apache.iotdb.udf.api.relational.table.argument.ScalarArgument;
+import org.apache.iotdb.udf.api.relational.table.argument.TableArgument;
+import org.apache.iotdb.udf.api.relational.table.processor.TableFunctionDataProcessor;
+import org.apache.iotdb.udf.api.relational.table.specification.ParameterSpecification;
+import org.apache.iotdb.udf.api.relational.table.specification.ScalarParameterSpecification;
+import org.apache.iotdb.udf.api.relational.table.specification.TableParameterSpecification;
+import org.apache.iotdb.udf.api.type.Type;
+
+import org.apache.tsfile.block.column.ColumnBuilder;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.iotdb.commons.udf.builtin.relational.tvf.WindowTVFUtils.findColumnIndex;
+
+public class CumulateTableFunction implements TableFunction {
+
+  private static final String DATA_PARAMETER_NAME = "DATA";
+  private static final String TIMECOL_PARAMETER_NAME = "TIMECOL";
+  private static final String SIZE_PARAMETER_NAME = "SIZE";
+  private static final String STEP_PARAMETER_NAME = "STEP";
+  private static final String ORIGIN_PARAMETER_NAME = "ORIGIN";
+
+  @Override
+  public List<ParameterSpecification> getArgumentsSpecifications() {
+    return Arrays.asList(
+        TableParameterSpecification.builder()
+            .name(DATA_PARAMETER_NAME)
+            .rowSemantics()
+            .passThroughColumns()
+            .build(),
+        ScalarParameterSpecification.builder()
+            .name(TIMECOL_PARAMETER_NAME)
+            .type(Type.STRING)
+            .defaultValue("time")
+            .build(),
+        ScalarParameterSpecification.builder().name(SIZE_PARAMETER_NAME).type(Type.INT64).build(),
+        ScalarParameterSpecification.builder().name(STEP_PARAMETER_NAME).type(Type.INT64).build(),
+        ScalarParameterSpecification.builder()
+            .name(ORIGIN_PARAMETER_NAME)
+            .type(Type.TIMESTAMP)
+            .defaultValue(0L)
+            .build());
+  }
+
+  @Override
+  public TableFunctionAnalysis analyze(Map<String, Argument> arguments) throws UDFException {
+    // size must be an integral multiple of step.
+    long size = (long) ((ScalarArgument) arguments.get(SIZE_PARAMETER_NAME)).getValue();
+    long step = (long) ((ScalarArgument) arguments.get(STEP_PARAMETER_NAME)).getValue();
+
+    if (size % step != 0) {
```

Comment:
Should check zero value in all table value function?

Reply:
Yes I think it is necessary to add a zero checker to avoid dead loop. HOPFunction also needs it.

Maybe we can add a Function<Object, Boolean> checker in ScalarParameterSpecification🤔?

Reply:
@Cpaulyz Any idea about this?

Reply:
Flink behavior:
HOP(..., slide => INTERVAL '5' MINUTES, SIZE => INTERVAL '0' MINUTES)
java.lang.IllegalArgumentException: SlidingWindowAssigner parameters must satisfy slide > 0 and size > 0

HOP(..., slide => INTERVAL '0' MINUTES, SIZE => INTERVAL '5' MINUTES)
java.lang.ArithmeticException: / by zero

囧. Anyway, it is better to add a zero checker. I'll try next week.